### DISCUSSION TITLE
How does LazyVim do language dependent tab spacing without ftplugin files

### DISCUSSION THREAD

**Discussion by roberte777** [LazyVim/LazyVim#2316] [General]
When I use lazyvim, Python spacing is 4 and ts spacing is 2. I can't find anywhere where this is set up. I thought you could only do this with ftplugin files. How does lazy set this up?

**Comment by dpetka2001** [comment]
There are ftplugin files distributed with the Neovim runtime files.  All these files [here](https://github.com/neovim/neovim/tree/master/runtime/ftplugin) get installed when you install Neovim.

**Reply by roberte777** [reply]
Thanks for the helpful response! I have a few follow up questions if you don't mind?

First, is there a good place to read more about these to answer any of the questions below? I've tried the help pages but I'm not really sure what to look for honestly. 

Do you have to do something to enable these, or do they get applied by default? 

I'm setting my tab widths in my settings. Does that override the ones defined in these files you linked? Folke sets his tab size to 2, but when testing in Python it's 4, so I would assume it does override your global settings?

The JavaScript file from the link you sent sets the tab size to 4, but when testing with LazyVim it's 2. How is this the case? 

Thank you again!!

**Reply by dpetka2001** [reply]
As far as I know, since those are `ftplugin` files, they are loaded per filetype of the buffer that is open. As for the order (without knowing too much about the intrinsics) my guess is that what user sets in his personal configuration overwrites those settings (but that also depends on the order of the files being loaded I believe).  So, that must be why in Python it's 4. About the JS it's being set in the LSP server itself (`tsserver`) and that's why it's 2.
